### PR TITLE
code improvement

### PR DESCRIPTION
**Pull request by n4-t3**
In my page when clicking view it leads to the wrong URL
 In create option page input field doesn't work
 Redirect to questions list page after question creation is complete
 Don't increase the options count if they add options to their own question

### CODE REVIEW

File: src/pages/class/[cid]/question/create.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default function Page() {
           similarOptions: [],
         })
 
-        push('/class/' + cid + '/question/' + res.data + '/create-option')
+        push('/class/' + cid)
```
Comment (jhw123) [inline src/pages/class/[cid]/question/create.tsx:66]: perhaps we should route to a specific topic
```suggestion
        push(`/class/${cid}?topic=${topic}`)
```

File: src/pages/api/loadUserActivity.ts [hunk 2]
```diff
@@ -25,15 +27,20 @@ export default apiController<LoadUserActivityParams, LoadUserActivityResults>(as
...
+
   const qOptions = await OptionModel.find({
     class: { $eq: cid },
     learningObjective: { $regex: topic, $options: 'i' },
     author: user.id,
   })
 
   if (qOptions) {
-    qOptionsNumber = qOptions.length
+    const filteredOptions = qOptions.filter(option => !generatedQStems.includes(option.qstem.toString()))
```
Comment (jhw123) [inline src/pages/api/loadUserActivity.ts:42]: what does this do?

**Review by jhw123 — APPROVED** [review]

File: src/pages/api/loadUserActivity.ts (hunk 2 above)
Comment (n4-t3) [reply src/pages/api/loadUserActivity.ts:42]: > what does this do?

I tried preventing option count from counting options that a user creates to their own question (the option count only counts options that a user creates for another user's question) to create more collaborative environment :)

File: src/pages/class/[cid]/question/create.tsx (hunk 1 above)
Comment (n4-t3) [reply src/pages/class/[cid]/question/create.tsx]: > perhaps we should route to a specific topic

Oh good idea! I have updated it here https://github.com/kixlab/kuiz/pull/60/commits/5787b8d1ce71efc5392211c5eb9bb68ccff0252a :)

File: src/pages/api/loadUserActivity.ts (hunk 2 above)
Comment (jhw123) [reply src/pages/api/loadUserActivity.ts:42]: I personally think (at least in the first deployment) we should control the point system through instructions. Let's ask Haesoo about this.

**Review by jhw123 — APPROVED** [review]

**Pull request merged**